Apps/W1/HybridBaseDeployment/app/src/codeunits/MigrationValidationMgmt.Codeunit.al

```diff
@@ -0,0 +1,345 @@
+namespace Microsoft.DataMigration;
+
+using System.Reflection;
+
+codeunit 40030 "Migration Validation Mgmt."
+{
+    /// <summary>
+    /// Start migration validation for the current company
+    /// </summary>
+    /// <param name="MigrationType">The type of migration</param>
+    /// <param name="Force">Force the validation. This would be done if conducting a manual validation.</param>
+    procedure StartValidation(MigrationType: Text; Force: Boolean)
+    begin
+        StartValidationSession(MigrationType, CompanyName(), false, Force);
+    end;
+
+    /// <summary>
+    /// Start migration validation for a specific company
+    /// </summary>
+    /// <param name="MigrationType">The type of migration</param>
+    /// <param name="Company">The company to be validated</param>
+    /// <param name="UseSession">Use a new session or not</param>
+    /// <param name="Force">Force the validation. This would be done if conducting a manual validation.</param>
+    procedure StartValidationSession(MigrationType: Text; Company: Text; UseSession: Boolean; Force: Boolean)
+    var
+        HybridCompanyStatus: Record "Hybrid Company Status";
+        MigrationValidatorRegistry: Record "Migration Validator Registry";
+        CompanyValidated: Boolean;
+        SessionId: Integer;
+    begin
+        MigrationValidatorRegistry.SetRange("Migration Type", MigrationType);
+
+        if Force then
+            DeleteMigrationValidationEntriesForCompany(Company)
+        else
+            MigrationValidatorRegistry.SetRange(Enabled, true);
+
+        if not MigrationValidatorRegistry.FindSet() then
+            exit;
+
+        Commit();
+        repeat
+            if UseSession then
+                Session.StartSession(SessionId, MigrationValidatorRegistry."Codeunit Id", Company)
+            else begin
+                Codeunit.Run(MigrationValidatorRegistry."Codeunit Id");
```

Comment:
There are multiple issues here. 
1. We cannot start many background sessions, it will consume too many resources. We can max start few sessions in parallel. 
2. If the Codeunit.Run throws an error we will not proceed with other tests. We need to be able to generically handle the errors.